source/data/tier1data/schemas/metadata-schemas.rst

```diff
@@ -0,0 +1,452 @@
+portal is that schemas that can be used to apply metadata cannot be
+modified. In other words, for a schema to be used in metadata annotation
+it must be fixed and stable; a schema that is undergoing changes cannot
+reliably be used for annotation. However, during the course of a
+research project it might be necessary to update a schema, and it would
+be impractical to create whole new schemas with different names for
+every such change. In order to tackle this issue, the ManGO portal
+implements a lifecycle via versioning and tags. In short, a schema can
+evolve and its evolutions can be registered as new versions of the same
+schema. Each version can have one of three statuses: (1) ‘draft’, while
+it is being designed and edited; (2) ‘published’, when it is ready to be
+implemented; and (3) ‘archived’, when it should not be used anymore,
+maybe because a new version has been published.
+
+In this context, the rest of this document will walk you through the
+process of creating, managing and applying metadata schemas. First,
+`Section 1 <#sec-draft>`__ will illustrate how to design a new schema
+from scratch and save a draft. Then, `Section 2 <#sec-lifecycle>`__ will
+briefly discuss the stages of a schema in more detail, including how
+they can be managed in the Metadata Schema Manager (the “Metadata
+schemas” tab in the ManGO portal). Finally,
+`Section 3 <#sec-application>`__ will show how a published schema can be
+used to annotate metadata.
+
+.. _sec-draft:
+
+1. Schema design
+================
+
+In order to design a new schema, we need to go to the “Metadata schemas”
+section via the left sidebar menu. This will first show a selection of
+projects (“realms”) to which your schemas may belong. After choosing
+one, the schema manager per se is shown (see `Figure 1 <#fig-start>`__):
+it has a button to create a new schema, under which any existing schemas
+are listed. Clicking on the button opens up a form on which the schema
+can be designed. As shown in `Figure 1 (b) <#fig-schemaeditor>`__, it
+includes a field to provide a name, one for a user-facing title and a
+button to add a field.
+
+.. container::
+   :name: figs-start
+
+   .. figure:: ../images/metadata-schemas/001-schema-start.png
+      :alt: Accordion with a list of realms, one of which is enframed in a blue square with a cursor icon on top. Underneath and connected via an arrow, an empty metadata schema manager that just has the title and a button that reads 'new schema'.
+      :width: 400
+      :name: fig-start-a
+
+      \(a) Choose a realm.
+
+   .. figure:: ../images/metadata-schemas/03-empty-schema.png
+      :alt: Initial form for an empty schema under the 'New schema' button, with fields to insert the schema ID and label, a button to add a new element and buttons to submit the form.
+      :width: 400
+      :name: fig-schemaeditor
+
+      \(b) An empty schema to start with.
+
+   Figure 1: A new schema from scratch.
```

Comment:
I see a few with this method of showing figures in containers:
* figures are not numbered automatically
* display is not flexible

We can enable `numfig` in our configuration to have the figures with captions to be numbered automatically. This also allows for simple referencing of figures in the text with ```:numref:`label````. Each figure will have its own number though, but I think it is a good tradeoff to simplify referencing the figures.

We can also replace the container with a grid that adapts to the width of the screen. In this way we can show figures side by side on larger screens.

See an example below for the first group of figures:
```suggestion
.. grid:: 1 1 2 2
   :gutter: 4

   .. grid-item::

      .. figure:: ../images/metadata-schemas/001-schema-start.png
         :alt: Accordion with a list of realms, one of which is enframed in a blue square with a cursor icon on top. Underneath and connected via an arrow, an empty metadata schema manager that just has the title and a button that reads 'new schema'.
         :width: 400
         :name: fig-start-a

         Choose a realm for new schema.

   .. grid-item::

      .. figure:: ../images/metadata-schemas/03-empty-schema.png
         :alt: Initial form for an empty schema under the 'New schema' button, with fields to insert the schema ID and label, a button to add a new element and buttons to submit the form.
         :width: 400
         :name: fig-schemaeditor

         An empty schema to start with.
         How to get your VSC account and access the different VSC services and platforms.
```

Enabling `numfig` in the configuration file can be done as follows:
```
diff --git a/source/conf.py b/source/conf.py
index 1e84b5c..c91bf47 100644
--- a/source/conf.py
+++ b/source/conf.py
@@ -85,6 +85,8 @@ exclude_patterns = []
 # To render double dashes in text as double dashes (and not as an "en dash"):
 smartquotes_action = 'qe'  # instead of default 'qDe'

+# Automatic numbering of figures
+numfig = True

 # -- Options for HTML output -------------------------------------------------

```

Reply:
Hello Alex, I have activated this configuration in conf.py, but the way of referencing you mentioned before doesn't work for me:

/home/jef/repositories/github/hpcleuven/VscDocumentation/source/data/tier1data/schemas/metadata-schemas.rst:76:Failed to create a cross reference. Any number is not assigned: fig-schemaeditor

Any idea what is going wrong?

Reply:
With regards to the grid:
With normal screens, as far as I tried, I need to set columns to 12 in order to make the images visible.
Otherwise, they are indeed side by side but too small.
Furthermore, it seems we lose the ability to reference a group of related pictures, since grids do not have a :name: property.

I

Reply:
I agree that the grid was not ideal. However, using figures with sub-figures is very confusing and the RST language is not prepared to handle those. In https://github.com/hpcleuven/VscDocumentation/pull/349/commits/d9a7a74a355840a4746f67152b75c9e01469a784 I reformatted it with simple figures and adding sub-sections to group those steps and figures that belong to the same _group_. Let me know if you like it, if not we can still iterate over it. :)